src/mpid/ch4/netmod/ofi/ofi_am_impl.h

```diff
@@ -297,13 +297,11 @@ static inline int MPIDI_OFI_am_isend_long(int rank,
     if (MPIDI_OFI_ENABLE_MR_SCALABLE) {
         /* Always allocates RMA ID from COMM_WORLD as the actual associated communicator
          * is not available here */
-        uint64_t index =
-            MPIDI_OFI_index_allocator_alloc(MPIDI_OFI_COMM
-                                            (MPIR_Process.comm_world).rma_id_allocator,
-                                            MPL_MEM_RMA);
-        MPIR_Assert(index < MPIDI_Global.max_huge_rmas);
-        lmt_info->rma_key = MPIDI_OFI_rma_key_pack(comm->context_id,
-                                                   MPIDI_OFI_KEY_TYPE_HUGE_RMA, index);
+        lmt_info->rma_key =
+            MPIDI_OFI_index_allocator_alloc(MPIDI_Global.mr_key_allocator, MPL_MEM_RMA);
+        MPIR_ERR_CHKANDSTMT(lmt_info->rma_key >=
+                            (((uint64_t) 1) << (MPIDI_Global.max_mr_key_size * 8)), mpi_errno,
```

Comment:
When `max_mr_key_size` equals 8, ) `((uint64_t) 1) << (MPIDI_Global.max_mr_key_size * 8)` actually equals `1`. Why are we checking if the `rma_key` in this way?

Reply:
I understand the check if to ensure the `rma_key` is not larger than the OFI MR key range which makes sence, but the case when the OFI MR key allows 64 bits, this check will be incorrect. I think I can fix it.